web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -112,27 +112,46 @@ function isR2L() {
     },
 
     make_multi: function() {
-        var $this = $(this),
-            all = $this.data('all'),
-            none = $this.data('none') || all,
-            allOpts = $this.data('all-options') || [],
-            extra = $this.data('extra'),
-            extraOpts = $this.data('extra-options') || [];
-
-            var presets = [{
-                name: all,
-                options: allOpts
-            }];
-
-            if (extra) {
-                presets.push({name: extra, options: extraOpts});
+        // A convenience wrapper around $.multiSelect() that translates HTML
+        // data-* attributes into settings for the multiSelect constructor.
+        var $select = $(this);
+        var settings = {};
+
+        if ( $select.data('none') ) {
+            settings.noneText = $select.data('none');
+        }
+
+        if ( $select.data('all') ) {
+            settings.allText = $select.data('all');
+
+            if ( ! settings.noneText ) {
+                settings.noneText = settings.allText;
             }
-        $this.multiSelect({
-            allText: all,
-            noneText: none,
-            positionMenuWithin: $('#side'),
-            presets: presets
-        });
+
+            if ( $select.data('all-options') ) {
+                settings.presets = [];
+                settings.presets.push({
+                    name: settings.allText,
+                    options: $select.data('all-options')
+                });
+            }
+        }
+
+        if ( $select.data('extra') && $select.data('extra-options') ) {
+            if ( typeof settings.presets === 'undefined' ) {
```

Comment:
This can just be `settings.presets = settings.presets || [];` I think. Or at least `if (!settings.presets)`, no need for typeof here.

Reply:
Yes, `settings.presets = settings.presets || [];` does the job. Not sure what I was thinking there!